src/pages/BrowserPage.qml

```diff
@@ -356,6 +356,12 @@ Page {
             bringToForeground(webView.chromeWindow)
             window.activate()
         }
+        onOpenSettingsRequested: {
+            pageStack.pop(browserPage, PageStackAction.Immediate)
```

Comment:
This is very bad that we need to pop from the stack as there might be web popups active. Just saying.